### DISCUSSION TITLE
Configurable parameter in Table Configuration with Declarative

### DISCUSSION THREAD

**Discussion by JL1829** [sqlalchemy/sqlalchemy#10581] [Usage Questions]
Hi: 

**SQLAlchemy Version: 2.0.20**
**Database: PostgreSQL 15.4**
**PGvector version: 0.2.3**

I have read through the documentation about the ORM for Table configuration with Declarative: https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html 

wondering is there any way to flexible to configure the parameter in the Table Class such as ``__tablename__`

For example: 

```python
import sqlalchemy
from pgvector.sqlalchemy import Vector
from sqlalchemy.dialects.postgresql import JSON, UUID


class EmbeddingStore(BaseModel):
    """Embedding store."""

    # is there any way pass in `__tablename__` on-the fly
    __tablename__ = "some_table_name_user_want_to_create"

    collection_id = sqlalchemy.Column(
        UUID(as_uuid=True),
        sqlalchemy.ForeignKey(
            f"{CollectionStore.__tablename__}.uuid",
            ondelete="CASCADE",
        ),
    )
    collection = relationship(CollectionStore, back_populates="embeddings")

    # here the `Vector(None)` the dimension can be pass in on-the-fly
    embedding: Vector = sqlalchemy.Column(Vector(None))
    document = sqlalchemy.Column(sqlalchemy.String, nullable=True)
    cmetadata = sqlalchemy.Column(JSON, nullable=True)

    # custom_id : any user defined id
    custom_id = sqlalchemy.Column(sqlalchemy.String, nullable=True)
```

The reason why behind this question is I am writing a PG Vector Store that need to be create table for different user or application, for example: 

```python
class SomeVectorStore:

    def __init__(self):
        balabala
    def __post__init__(self):
        table = EmbeddingStore(
                tablename = "some fancy name"
                vector_dim = 12345
        )
        Base.metadata.create_all(self.connection)
```

Thanks very much

**Comment by CaselIT** [comment]
Hi,

If you want to use orm you can look into https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.declared_attr

It can be used either with a mixin or with an abstract class https://docs.sqlalchemy.org/en/20/orm/inheritance.html#abstract-concrete-classes
